uber/utils.py

```diff
@@ -825,24 +825,27 @@ def get_unpaid_promo_code_uses_count(cls, id, already_counted_attendee_ids=None)
         return promo_code_count
 
     @classmethod
-    def to_sessionized(cls, m):
+    def to_sessionized(cls, m, name='', badges=0):
         from uber.models import Attendee, Group
         if is_listy(m):
             return [cls.to_sessionized(t) for t in m]
         elif isinstance(m, dict):
             return m
         elif isinstance(m, Attendee):
-            return m.to_dict(
+            d = m.to_dict(
                 Attendee.to_dict_default_attrs
                 + ['promo_code']
                 + list(Attendee._extra_apply_attrs_restricted))
+            d['name'] = name
```

Comment:
This implementation is kind of janky due to the nature of promo code groups -- as far as money goes, the Attendee object needs to be the one who pays for a promo code group, but we still need to sessionize and desessionize certain group-specific info. We're doing that here by hijacking the Attendee dict, which can hold extra variables.

The real fix for this is to implement a cart-like system and make PromoCodeGroups a piece of merch. Sadly, I did not have time to do this.